File: class2relational/src/main/java/org/atlanmod/model/dynamic/relationalModel/RelationalLink.java
package org.atlanmod.model.dynamic.relationalModel;

import org.atlanmod.model.DynamicElement;
import org.atlanmod.model.DynamicLink;

import java.util.List;

public abstract class RelationalLink extends DynamicLink {

    public RelationalLink(String type, DynamicElement source, List<DynamicElement> target) {
        super(type, source, target);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof RelationalLink){
            RelationalLink obj_ = (RelationalLink) obj;
            return  this.getType().equals(obj_.getType())
                    && this.getSource().equals(obj_.getSource())
                    && this.getTarget().equals(obj_.getTarget());
        }
        return false;
    }

}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/relationalmodel/RelationalLink.scala
package org.atlanmod.class2relational.model.relationalmodel

import org.atlanmod.tl.model.impl.dynamic.DynamicLink
import org.atlanmod.tl.util.ListUtils

abstract class RelationalLink (type_ : String, source: RelationalElement, target: List[RelationalElement])
  extends DynamicLink(type_, source, target){

    override def equals(o: Any): Boolean = {
        o match {
            case obj: RelationalLink => type_.equals(obj.getType) & source.equals(obj.getSource) &
              ListUtils.eqList(target, obj.getTarget)
            case _ => false
        }
    }

    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: RelationalLink =>
                source.weak_equals(obj.getSource) &
                  type_.equals(obj.getType) &
                  ListUtils.weak_eqList(obj.getTarget, target)
            case _ => false
        }
    }
}